### ISSUE TITLE
TST: failing to load `pyart` 

### ISSUE THREAD

**Issue by neutrinoceros** [yt-project/yt#5041]
The test pipeline on Jenkins appears to be broken as an `ImportError` is raised when trying to lazy-load pyart

https://tests.yt-project.org/job/yt_py310_git/8571/

I have reported the problem upstream https://github.com/ARM-DOE/pyart/issues/1680
In the meantime we can add `datatree` as an optional requirement on our side as a temporary measure.

**Comment by kmuehlbauer** [comment]
@neutrinoceros Heads-up, datatree was merged into [xarray codebase](https://github.com/pydata/xarray/blob/main/DATATREE_MIGRATION_GUIDE.md) recently.

To use xarray-datatree (as in the test) you would also need to pin xarray to <=2024.9.0 and xradar <=to 0.7.0.

xradar 0.7.0 has a pin for xarray <= 2024.9.0
xradar 0.8.0 has a pin for xarray >= 2024.10.0

HTH

**Comment by neutrinoceros** [comment]
marking this as a blocker since we cannot merge anything until there's at least a workaround for this issue

**Comment by neutrinoceros** [comment]
fixed in #5042
